pkg/azure/client.go

```diff
@@ -120,22 +132,80 @@ func (c *azureDNS) Delete(ctx context.Context, record *phonebook.DNSRecord) erro
 
 // Convert a DNSRecord to an Azure DNS record set
 func (c *azureDNS) resourceRecordSet(record *phonebook.DNSRecord) armdns.RecordSet {
+	ttl := defaultTTL
+	if record.Spec.TTL != nil {
+		ttl = *record.Spec.TTL
+	}
+
 	params := armdns.RecordSet{
 		Properties: &armdns.RecordSetProperties{
-			TTL: to.Ptr(int64(60)),
+			TTL: to.Ptr(int64(ttl)),
 		},
 	}
 
-	// Create specific record types based on the DNS type (e.g., A, CNAME)
+	// Create specific record types based on the DNS type
 	switch armdns.RecordType(record.Spec.RecordType) {
 	case armdns.RecordTypeA:
-		params.Properties.ARecords = []*armdns.ARecord{
-			{IPv4Address: to.Ptr(record.Spec.Targets[0])},
+		aRecords := make([]*armdns.ARecord, len(record.Spec.Targets))
+		for i, target := range record.Spec.Targets {
+			aRecords[i] = &armdns.ARecord{IPv4Address: to.Ptr(target)}
+		}
+		params.Properties.ARecords = aRecords
+
+	case armdns.RecordTypeAAAA:
+		aaaaRecords := make([]*armdns.AaaaRecord, len(record.Spec.Targets))
+		for i, target := range record.Spec.Targets {
+			aaaaRecords[i] = &armdns.AaaaRecord{IPv6Address: to.Ptr(target)}
 		}
+		params.Properties.AaaaRecords = aaaaRecords
+
 	case armdns.RecordTypeCNAME:
+		// CNAME can only have one target
 		params.Properties.CnameRecord = &armdns.CnameRecord{
 			Cname: to.Ptr(record.Spec.Targets[0]),
 		}
+
+	case armdns.RecordTypeMX:
+		mxRecords := make([]*armdns.MxRecord, len(record.Spec.Targets))
+		for i, target := range record.Spec.Targets {
+			parts := strings.SplitN(target, " ", 2)
+			if len(parts) != 2 {
+				continue // Skip invalid entries
```

Comment:
Returning an error here the better option here. I rather have the whole DNS Record fail with a meaningful message.

The Error will show up in the `events` of the DNS Record as well as within the Condition.

Reply:
Done :)